_summaries/he2019momentum_1.md

```diff
@@ -1,19 +1,87 @@
 ---
 layout: summary
 title: Summary
-paper: {{paper_tag}}
+paper: he2019momentum
 # Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author: elampietti
+score: 9
 ---
 
-TODO: Summarize the paper:
-* What is the core idea?
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+Traditionally, computer vision tasks are more effective with supervised pre-training while tasks in natural language processing (NLP) are successful with unsupervised representation learning.
+This is because NLP tasks build tokenized dictionaries in their discrete signal spaces that unsupervised learning can be applied to.
+However, computer vision has a continuous, high-dimensional space that makes it more challenging for dictionary building.
+This paper focuses on how to build dynamic dictionaries for unsupervised visual representation learning using their Momentum Constrast (MoCo) for training.
+
+The keys in this dynamic dictionary are images or patches and are put into an encoder network allowing dictionary look-up for a matching key.
+The goal of this learning would be to minimize contrastive loss when training the visual representation encoder.
+This training process is visualized in the below figure.
+
+![train_encoder](https://user-images.githubusercontent.com/7085644/140272389-b0ec71d8-e51e-4dbd-a67f-1ba9607cac21.PNG)
+
+The optimal dynamic dictionary should be large to represent the continuous, high-dimensional visual space and they should also be consistent.
+The dictionary has a queue of samples where the current mini-batch encodings are enqueued while the older are dequeued.
+Consistency is then maintained with a key encoder that slowly progressed as a momentum-based moving average of the query encoder.
+
+Contrastive learning is used to train the encoder for a dictionary look-up task.
+This is done by finding the key that has the smallest contrastive loss with the query, meaning it is the most similar.
+This similarity loss function is shown below.
+
+![contrastive_loss](https://user-images.githubusercontent.com/7085644/140273552-d1d51632-da7f-4f34-bdd8-19a91c9e7cc0.PNG)
+
+Momentum Contrast is employed to make sure that the dynamic dictionary being built is large and the encoder of the keys is consistent during its evolution.
+This is done by using a queue for the dictionary data samples so that the dictionary size can be larger than the mini-batch size and set as a hyper-parameter.
+A momentum update is then employed during back-propagation to solve the issue for updating keys when using a queue for samples.
+This means that the encoder will evolve slowly, therefore keeping it consistent.
```

Comment:
For this section, I would maybe expand each subsection (dictionary as  a queue, momentum update) a bit and give some more description on what is actually being done.

Reply:
Maybe also add some content from "Relations to previous mechanisms"